src/contexts/Theme.tsx

```diff
@@ -1,10 +1,15 @@
 import { ChakraProvider } from "@chakra-ui/react";
 import type { FunctionComponent } from "react";
-import { theme } from "../theme";
+import { PageLoader } from "../components/PageLoader";
+import { makeTheme } from "../theme";
+import { useConfig } from "./Config";
 
 export const Theme: FunctionComponent = ({ children }) => {
-  return (
-    <ChakraProvider resetCSS theme={theme}>
+  const { loading, data } = useConfig();
+  return loading ? (
+    <PageLoader />
```

Comment:
IMO, we should move the UI providers such as `Theme`, `ExternalLinkWarning`, and `CardView` (to be deleted) into `App.tsx` where we could just do a component level `if (configLoading) return <PageLoader />` so we can avoid importing PageLoader into multiple files

Reply:
Or actually on second thought, we could just allow `makeTheme`'s data parameter to be optional and let it update when `data` updates since it's all hooked into react context.